src/os/bluestore/BlueStore.cc

```diff
@@ -11517,12 +11517,17 @@ void BlueStore::_do_write_small(
     prev_ep = end; // to avoid this extent check as it's a duplicate
   }
 
+  set<const bluestore_blob_t*> inspected_blobs;
```

Comment:
i wonder if we can do something slightly clever here to avoid the std::set.  maybe flat_set, and stop looking/maintaining it once it hits the max blob threshold?  originally i was hoping we could just see when the blobptr changed value (``if (b->get_blob() != last_blob) { last_blob = b->get_blob(); ++count; }`` and count that, but that isn't counting quite the same thing.  although maybe that is good enough?  in the worst case it'll be off by a factor of two i think?

Reply:
I doubt tracking blobptr change is good enough approach - imagine two "interleaving" (the first extent points to blob1, the second - to blob2, the third to blob1, etc). Actually we have just two 2 blobs but estimated count might be up to max_extent_count (=max_blob_size / 2).
So replaced with flat_set with preallocated capacity. And stopping insertions on threshold (=capacity) hit. 
Generally I don't think all this stuff will have significant impact on the performance since it applies to small writes only and blob set threshold is pretty small (up to 16 entries for spinners = max_blob_size / min_alloc_size * 2).